### PR TITLE
download token validation & BSR closes 21

### PR DESCRIPTION
**Pull request by Dromadon**

### CODE REVIEW

File: app/hallebarde/download_authorizer.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import logging
+
+from hallebarde.infrastructure import exchange_repository
+from hallebarde.responses import authorizer_responses
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def handle(event: dict, context) -> dict:
```
Comment (Mehdi-H) [inline app/hallebarde/download_authorizer.py]:
```suggestion
def handle(event: dict, context: dict) -> dict:
```

File: app/hallebarde/responses/authorizer_responses.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+def generate_policy(is_valid: bool, resource: str) -> dict:
```
Comment (Mehdi-H) [inline app/hallebarde/responses/authorizer_responses.py]: I would call it `generate_aws_policy` to enforce the fact that this format comes from AWS and it is mandatory to respect it

File: app/hallebarde/responses/authorizer_responses.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    return {
```
Comment (Mehdi-H) [inline app/hallebarde/responses/authorizer_responses.py]: Also, did you look at this https://boto3.amazonaws.com/v1/documentation/api/latest/guide/iam-example-policies.html ? For AWS policy generation with boto3

Don't know if it is what we are doing here or totally something else

File: app/hallebarde/download_authorizer.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    policy = authorizer_responses.generate_policy(validate(token), method_arn)
+    return authorizer_responses.generate_response('hallebarde_download_authorizer', policy)
+
+
+def validate(upload_token: str) -> bool:
+    exchange = exchange_repository.get_by_download_token(upload_token)
+    return True if exchange else False
+
+
```
Comment (Mehdi-H) [inline app/hallebarde/download_authorizer.py]:
```suggestion
```

Comment (Mehdi-H) [reply app/hallebarde/download_authorizer.py]: From flake8: `app/hallebarde/download_authorizer.py:23:1: W391 blank line at end of file`
You can find this feedback here, in the github actions CI pipeline: https://github.com/Dromadon/hallebarde/pull/28/checks?check_run_id=506458094#step:6:23

File: app/hallebarde/upload_authorizer.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import logging
+
+from hallebarde.infrastructure import exchange_repository
+from hallebarde.responses import authorizer_responses
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def handle(event: dict, context) -> dict:
```
Comment (Mehdi-H) [inline app/hallebarde/upload_authorizer.py]:
```suggestion
def handle(event: dict, context: dict) -> dict:
```

File: app/tests/infrastructure/test_exchange_repository.py [hunk 6]
```diff
@@ -145,3 +145,26 @@ def test_get_by_upload_token_should_return_none_if_no_exchange_exists(self, mock
 
         # Then
         assert actual_exchange is None
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_by_download_token_should_return_exchange(self, mock_get_table, get_dynamodb_table, an_exchange):
```
Comment (Mehdi-H) [inline app/tests/infrastructure/test_exchange_repository.py]:
```suggestion
    def test_get_by_download_token_should_return_an_exchange(self, mock_get_table, get_dynamodb_table, an_exchange):
```

File: app/tests/responses/test_authorizer_responses.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            }]
+        }
+
+    def test_authorizer_should_return_an_authentication_with_an_id_and_a_policy(self):
+        # When
+        auth_response: dict = generate_response("aprincipalid", {})
+
+        # Then
+        assert 'principalId' in auth_response.keys()
+        assert 'policyDocument' in auth_response.keys()
```
Comment (Mehdi-H) [inline app/tests/responses/test_authorizer_responses.py]:
```suggestion
        assert 'policyDocument' in auth_response.keys()

```

File: app/hallebarde/responses/authorizer_responses.py (hunk 3 above)
Comment (Dromadon) [reply app/hallebarde/responses/authorizer_responses.py]: Unfortunately it is not. This is the API for CRUD on IAM policies, but from what I understand it does not provide any utilities for writing such policies (which can be written as dicts in python)

**Pull request merged**